### ISSUE TITLE
GpgFrontend 2.1.4 for Linux doesn't give any option to create BrainPool keys.

### ISSUE THREAD

**Issue by shamsuddin82** [saturneric/GpgFrontend#175]
The windows version clearly give option to create ECDSA BrainPool for Key generation and ECDH BrainPool for Subkey generation. But the linux version doesn't give such option. Although this option was available in 2.1.3, but not any more.

**Comment by saturneric** [comment]
Please check the version of gnupg installed on your Linux system. If it is less than 2.3.0, some key algorithms will not appear.

**Comment by shamsuddin82** [comment]
> Please check the version of gnupg installed on your Linux system. If it is less than 2.3.0, some key algorithms will not appear.

It is lesser than 2.3 but if this is the case then why this option was available in gpgfrontend 2.1.3?

**Comment by saturneric** [comment]
I tested these algorithms in the later stages of the 2.1.4 development. On my machine, the gnupg version was 2.2.40. When I tested the Brainpool algorithm, I found a bug in gpg at this time: it could not infer whether to use ECDSA or ECDH from the type and purpose of the key, which prevented users from using GpgFrontend to generate Brainpool algorithm keys (including subkeys). 

However, after testing, the same parameters worked properly in 2.4.x. However, I spent a long time searching. I found [a ticket](https://dev.gnupg.org/T5444), but the development team member said that this worked in 2.4.x, but did not provide commit information about the fix. So I couldn't locate which version of gnupg fixed this bug. In the end, I could only increase the version requirement for the extended algorithms in GpgFrontend, and I set this value to 2.3.0, which is the gnupg version generating ECC algorithm keys by default. 

And after 2.3.0, gnupg provides support for more algorithms, such as X448, Ed448, it seems ready to meet the challenges brought by various ECC key algorithms. So I can only speculate that this problem will not occur in most machines under this version.
